test/rat.lean

```diff
@@ -0,0 +1,11 @@
+import data.rat.meta
+
+run_cmd guard $ (reflect (3/15 : ℚ) : expr) = `((3/15 : ℚ))
```

Comment:
Something is fishy here. Shouldn't the result be `` `((1/5 : ℚ))``, because `3/15 = 1/5`? But also, the test passes even without importing `data.rat.meta` (just `data.rat.basic`). So it must not be using the new instance of `has_reflect ℚ`, but then I'm not sure what it is doing instead.

I'm now really confused about the relationship between `has_reflect`, `reflect` and the `` `(a)`` notation. Is the intent for the new `has_reflect ℚ` instance to override an implicitly generated one? This seems kind of fragile...

Reply:
Ah, thanks for the explanation. I didn't do enough experiments to figure out that reflection has this special case.

Reply:
Yeah, unfortunately this is written down nowhere and I need to relearn it every time I do something related.